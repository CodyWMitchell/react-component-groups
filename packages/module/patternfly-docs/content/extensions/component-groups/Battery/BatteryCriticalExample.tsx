import React from 'react';
import { Battery } from '@patternfly/react-component-groups';

const BatteryCriticalExample: React.FunctionComponent = () => (
  <>
    <Battery label="With prop: 4" severity={4} />
    <Battery className="pf-u-ml-md" label="With prop: critical" severity="critical" />
  </>
);

export default BatteryCriticalExample;

