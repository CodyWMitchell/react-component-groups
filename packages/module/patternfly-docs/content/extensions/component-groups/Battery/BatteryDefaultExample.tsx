import React from 'react';
import { Battery } from '@patternfly/react-component-groups';

const BatteryDefaultExample: React.FunctionComponent = () => <Battery label="Default" severity={'an unknown value' as any}/>

export default BatteryDefaultExample;
