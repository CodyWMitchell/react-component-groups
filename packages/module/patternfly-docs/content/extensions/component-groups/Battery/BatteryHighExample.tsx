import React from 'react';
import { Battery } from '@patternfly/react-component-groups';

const BatteryHighExample: React.FunctionComponent = () => (
  <>
    <Battery label="With prop: 3" severity={3} />
    <Battery className="pf-u-ml-md" label="With prop: high" severity="high" />
    <Battery className="pf-u-ml-md" label="With prop: error" severity="error" />
  </>
);

export default BatteryHighExample;