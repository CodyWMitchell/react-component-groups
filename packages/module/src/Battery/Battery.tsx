import React from 'react';
import classNames from 'classnames';
import CriticalBattery from './CriticalBattery';
import HighBattery from './HighBattery';
import MediumBattery from './MediumBattery';
import LowBattery from './LowBattery';
import NullBattery from './NullBattery';
import { createUseStyles } from 'react-jss'

const batteryDefault = {
  '& svg': {
    '& path': { fill: 'var(--pf-v5-global--disabled-color--200)' }
  }
};

const batteryLow = {
  '& svg': {
    '& path': { fill: 'var(--pf-v5-global--success-color--100)' }
  }
};

const batteryMedium = {
  '& svg': {
    '& path': { fill: 'var(--pf-v5-global--warning-color--100)' }
  }
};

const batteryHigh = {
  '& svg': {
    '& path': { fill: 'var(--pf-v5-global--palette--orange-300)' }
  }
};

const batteryCritical = {
  '& svg': {
    '& path': { fill: 'var(--pf-v5-global--danger-color--100)' }
  }
};

const useStyles = createUseStyles({
  battery: {
    display: 'inline-block',
    'line-height': 0,
    '& svg': {
      position: 'relative',
      top: 'var(--pf-v5-global--spacer--sm)',
      height: '1.75rem'
    }
  },

  'battery-0': batteryDefault,
  'battery-null': batteryDefault,

  'battery-1': batteryLow,
  'battery-low': batteryLow,
  'battery-info': batteryLow,

  'battery-2': batteryMedium,
  'battery-medium': batteryMedium,
  'battery-warn': batteryMedium,

  'battery-3': batteryHigh,
  'battery-high': batteryHigh,
  'battery-error': batteryHigh,

  'battery-4': batteryCritical,
  'battery-critical': batteryCritical,
});

export type BatterySeverity = 1 | 2 | 3 | 4 | 'info' | 'low' | 'warn' | 'medium' | 'error' | 'high' | 'critical';

export interface BatteryProps extends React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement> {
  /** Determines a variant of displayed Battery component */
  severity: BatterySeverity;
  /** Label displayed next to the battery */
  label: string;
  /** Option to hide the label */
  labelHidden?: boolean;
  /** Custom className */
  className?: string;
}

export const Battery: React.FunctionComponent<BatteryProps> = ({ severity, label, labelHidden, className, ...props }: BatteryProps) => {
  const classes = useStyles();
  const batteryClasses = classNames(classes.battery, classes[`battery-${severity}`], className);

  let ariaLabels = {};
  if (labelHidden) {
    ariaLabels = { ['aria-label']: `${severity} ${label}` };
  }

  const batteryLevels = (severity: BatterySeverity) => {
    switch (severity) {
    case 'critical':
    case 4:
      return <CriticalBattery />;
    case 'high':
    case 'error':
    case 3:
      return <HighBattery />;
    case 'medium':
    case 'warn':
    case 2:
      return <MediumBattery />;
    case 'low':
    case 'info':
    case 1:
      return <LowBattery />;
    default:
      // eslint-disable-next-line
        console.error('Warning: Unsupported value presented to battery component');
      return <NullBattery />;
    }
  };

  return (
    <React.Fragment>
      {/* eslint-disable-next-line react/no-unknown-property */}
      <i className={batteryClasses} {...ariaLabels} {...props} widget-type="InsightsBattery" widget-id={label}>
        <svg
          version="1.1"
          x="0px"
          y="0px"
          viewBox="0 0 448 512"
          style={{ enableBackground: 'new 0 0 448 512' } as React.CSSProperties}
          shapeRendering="geometricpresision"
        >
          <path
            style={{
              fill: 'none',
              fillOpacity: 1,
              stroke: '#969696',
              strokeWidth: 41.96378708,
              strokeLinejoin: 'round',
              strokeMiterlimit: 4,
              strokeDasharray: 'none',
              strokeDashoffset: 0,
              strokeOpacity: 1,
            }}
            d="m 144.16452,21.032222 h 159.67454 q 123.1748,0 123.1748,128.667868 v 212.64759 q 0,128.66788 -123.1748,128.66788 H 144.16452 q -123.174811,0 -123.174811,-128.66788 V 149.70009 q 0,-128.667868 123.174811,-128.667868 z"
          />
          {batteryLevels(severity)}
        </svg>
      </i>
      {!labelHidden && <span className="label"> {label} </span>}
    </React.Fragment>
  );
};

export default Battery;
