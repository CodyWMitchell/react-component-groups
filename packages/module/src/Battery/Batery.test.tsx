import React from 'react';
import { render } from '@testing-library/react';
import Battery, { BatterySeverity } from './Battery';

describe('Battery component', () => {
  jest.spyOn(global.console, 'error');
  describe('should render correctly', () => {
    ([ 'critical', 4 ] as BatterySeverity[]).forEach((severity) => {
      test(`CriticalBattery - ${severity}`, () => {
        const { container } = render(<Battery severity={severity as BatterySeverity} label={`${severity}`} />);
        expect(container).toMatchSnapshot();
      });
    });

    ([ 'high', 'error', 3 ] as BatterySeverity[]).forEach((severity) => {
      test(`HighBattery - ${severity}`, () => {
        const { container } = render(<Battery severity={severity as BatterySeverity} label={`${severity}`} />);
        expect(container).toMatchSnapshot();
      });
    });

    ([ 'medium', 'warn', 2 ] as BatterySeverity[]).forEach((severity) => {
      test(`MediumBattery - ${severity}`, () => {
        const { container } = render(<Battery severity={severity as BatterySeverity} label={`${severity}`} />);
        expect(container).toMatchSnapshot();
      });
    });

    ([ 'low', 'info', 1 ] as BatterySeverity[]).forEach((severity) => {
      test(`LowBattery - ${severity}`, () => {
        const { container } = render(<Battery severity={severity} label={`${severity}`} />);
        expect(container).toMatchSnapshot();
      });
    });

    test('NullBatery, default', () => {
      const { container } = render(<Battery severity={'' as BatterySeverity} label={''} />);
      expect(container).toMatchSnapshot();
      // eslint-disable-next-line no-console
      expect(console.error).toBeCalled();
    });
  });

  describe('API', () => {
    test('should hide label', () => {
      const { container } = render(<Battery severity={'high'} label={'high'} labelHidden />);
      expect(container).toMatchSnapshot();
    });
  });
});
