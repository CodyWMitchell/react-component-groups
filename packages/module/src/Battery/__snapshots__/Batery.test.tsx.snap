// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Battery component API should hide label 1`] = `
<div>
  <i
    aria-label="high high"
    class="battery-0-2-1 batteryHigh-0-2-5"
    widget-id="high"
    widget-type="Battery"
  >
    <svg
      shape-rendering="geometricpresision"
      style="enable-background: new 0 0 448 512;"
      version="1.1"
      viewBox="0 0 448 512"
      x="0px"
      y="0px"
    >
      <path
        d="m 144.16452,21.032222 h 159.67454 q 123.1748,0 123.1748,128.667868 v 212.64759 q 0,128.66788 -123.1748,128.66788 H 144.16452 q -123.174811,0 -123.174811,-128.66788 V 149.70009 q 0,-128.667868 123.174811,-128.667868 z"
        style="fill: none; fill-opacity: 1; stroke: #969696; stroke-width: 41.96378708; stroke-linejoin: round; stroke-miterlimit: 4; stroke-dasharray: none; stroke-dashoffset: 0; stroke-opacity: 1;"
      />
      <path
        d="M 99.168857,235.25069 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z M 99.168857,327.14542 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.19999 v -28 c 0,-6.16001 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.19999 9.698857,11.19999 z M 99.168993,419.0375 H 351.33927 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168993 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z"
      />
    </svg>
  </i>
</div>
`;

exports[`Battery component should render correctly CriticalBattery - 4 1`] = `
<div>
  <i
    aria-label="4 4"
    class="battery-0-2-1 batteryCritical-0-2-6"
    widget-id="4"
    widget-type="Battery"
  >
    <svg
      shape-rendering="geometricpresision"
      style="enable-background: new 0 0 448 512;"
      version="1.1"
      viewBox="0 0 448 512"
      x="0px"
      y="0px"
    >
      <path
        d="m 144.16452,21.032222 h 159.67454 q 123.1748,0 123.1748,128.667868 v 212.64759 q 0,128.66788 -123.1748,128.66788 H 144.16452 q -123.174811,0 -123.174811,-128.66788 V 149.70009 q 0,-128.667868 123.174811,-128.667868 z"
        style="fill: none; fill-opacity: 1; stroke: #969696; stroke-width: 41.96378708; stroke-linejoin: round; stroke-miterlimit: 4; stroke-dasharray: none; stroke-dashoffset: 0; stroke-opacity: 1;"
      />
      <path
        d="M 99.168858,143.38516 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698858,11.2 z M 99.168857,235.25069 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z M 99.168857,327.14542 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.19999 v -28 c 0,-6.16001 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.19999 9.698857,11.19999 z M 99.168993,419.0375 H 351.33927 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168993 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z"
      />
    </svg>
  </i>
  <span>
     
    4
     
  </span>
</div>
`;

exports[`Battery component should render correctly CriticalBattery - critical 1`] = `
<div>
  <i
    aria-label="critical critical"
    class="battery-0-2-1 batteryCritical-0-2-6"
    widget-id="critical"
    widget-type="Battery"
  >
    <svg
      shape-rendering="geometricpresision"
      style="enable-background: new 0 0 448 512;"
      version="1.1"
      viewBox="0 0 448 512"
      x="0px"
      y="0px"
    >
      <path
        d="m 144.16452,21.032222 h 159.67454 q 123.1748,0 123.1748,128.667868 v 212.64759 q 0,128.66788 -123.1748,128.66788 H 144.16452 q -123.174811,0 -123.174811,-128.66788 V 149.70009 q 0,-128.667868 123.174811,-128.667868 z"
        style="fill: none; fill-opacity: 1; stroke: #969696; stroke-width: 41.96378708; stroke-linejoin: round; stroke-miterlimit: 4; stroke-dasharray: none; stroke-dashoffset: 0; stroke-opacity: 1;"
      />
      <path
        d="M 99.168858,143.38516 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698858,11.2 z M 99.168857,235.25069 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z M 99.168857,327.14542 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.19999 v -28 c 0,-6.16001 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.19999 9.698857,11.19999 z M 99.168993,419.0375 H 351.33927 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168993 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z"
      />
    </svg>
  </i>
  <span>
     
    critical
     
  </span>
</div>
`;

exports[`Battery component should render correctly HighBattery - 3 1`] = `
<div>
  <i
    aria-label="3 3"
    class="battery-0-2-1 batteryHigh-0-2-5"
    widget-id="3"
    widget-type="Battery"
  >
    <svg
      shape-rendering="geometricpresision"
      style="enable-background: new 0 0 448 512;"
      version="1.1"
      viewBox="0 0 448 512"
      x="0px"
      y="0px"
    >
      <path
        d="m 144.16452,21.032222 h 159.67454 q 123.1748,0 123.1748,128.667868 v 212.64759 q 0,128.66788 -123.1748,128.66788 H 144.16452 q -123.174811,0 -123.174811,-128.66788 V 149.70009 q 0,-128.667868 123.174811,-128.667868 z"
        style="fill: none; fill-opacity: 1; stroke: #969696; stroke-width: 41.96378708; stroke-linejoin: round; stroke-miterlimit: 4; stroke-dasharray: none; stroke-dashoffset: 0; stroke-opacity: 1;"
      />
      <path
        d="M 99.168857,235.25069 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z M 99.168857,327.14542 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.19999 v -28 c 0,-6.16001 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.19999 9.698857,11.19999 z M 99.168993,419.0375 H 351.33927 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168993 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z"
      />
    </svg>
  </i>
  <span>
     
    3
     
  </span>
</div>
`;

exports[`Battery component should render correctly HighBattery - error 1`] = `
<div>
  <i
    aria-label="error error"
    class="battery-0-2-1 batteryHigh-0-2-5"
    widget-id="error"
    widget-type="Battery"
  >
    <svg
      shape-rendering="geometricpresision"
      style="enable-background: new 0 0 448 512;"
      version="1.1"
      viewBox="0 0 448 512"
      x="0px"
      y="0px"
    >
      <path
        d="m 144.16452,21.032222 h 159.67454 q 123.1748,0 123.1748,128.667868 v 212.64759 q 0,128.66788 -123.1748,128.66788 H 144.16452 q -123.174811,0 -123.174811,-128.66788 V 149.70009 q 0,-128.667868 123.174811,-128.667868 z"
        style="fill: none; fill-opacity: 1; stroke: #969696; stroke-width: 41.96378708; stroke-linejoin: round; stroke-miterlimit: 4; stroke-dasharray: none; stroke-dashoffset: 0; stroke-opacity: 1;"
      />
      <path
        d="M 99.168857,235.25069 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z M 99.168857,327.14542 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.19999 v -28 c 0,-6.16001 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.19999 9.698857,11.19999 z M 99.168993,419.0375 H 351.33927 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168993 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z"
      />
    </svg>
  </i>
  <span>
     
    error
     
  </span>
</div>
`;

exports[`Battery component should render correctly HighBattery - high 1`] = `
<div>
  <i
    aria-label="high high"
    class="battery-0-2-1 batteryHigh-0-2-5"
    widget-id="high"
    widget-type="Battery"
  >
    <svg
      shape-rendering="geometricpresision"
      style="enable-background: new 0 0 448 512;"
      version="1.1"
      viewBox="0 0 448 512"
      x="0px"
      y="0px"
    >
      <path
        d="m 144.16452,21.032222 h 159.67454 q 123.1748,0 123.1748,128.667868 v 212.64759 q 0,128.66788 -123.1748,128.66788 H 144.16452 q -123.174811,0 -123.174811,-128.66788 V 149.70009 q 0,-128.667868 123.174811,-128.667868 z"
        style="fill: none; fill-opacity: 1; stroke: #969696; stroke-width: 41.96378708; stroke-linejoin: round; stroke-miterlimit: 4; stroke-dasharray: none; stroke-dashoffset: 0; stroke-opacity: 1;"
      />
      <path
        d="M 99.168857,235.25069 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z M 99.168857,327.14542 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.19999 v -28 c 0,-6.16001 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.19999 9.698857,11.19999 z M 99.168993,419.0375 H 351.33927 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168993 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z"
      />
    </svg>
  </i>
  <span>
     
    high
     
  </span>
</div>
`;

exports[`Battery component should render correctly LowBattery - 1 1`] = `
<div>
  <i
    aria-label="1 1"
    class="battery-0-2-1 batteryLow-0-2-3"
    widget-id="1"
    widget-type="Battery"
  >
    <svg
      shape-rendering="geometricpresision"
      style="enable-background: new 0 0 448 512;"
      version="1.1"
      viewBox="0 0 448 512"
      x="0px"
      y="0px"
    >
      <path
        d="m 144.16452,21.032222 h 159.67454 q 123.1748,0 123.1748,128.667868 v 212.64759 q 0,128.66788 -123.1748,128.66788 H 144.16452 q -123.174811,0 -123.174811,-128.66788 V 149.70009 q 0,-128.667868 123.174811,-128.667868 z"
        style="fill: none; fill-opacity: 1; stroke: #969696; stroke-width: 41.96378708; stroke-linejoin: round; stroke-miterlimit: 4; stroke-dasharray: none; stroke-dashoffset: 0; stroke-opacity: 1;"
      />
      <path
        d="M 99.168993,419.0375 H 351.33927 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168993 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z"
      />
    </svg>
  </i>
  <span>
     
    1
     
  </span>
</div>
`;

exports[`Battery component should render correctly LowBattery - info 1`] = `
<div>
  <i
    aria-label="info info"
    class="battery-0-2-1 batteryLow-0-2-3"
    widget-id="info"
    widget-type="Battery"
  >
    <svg
      shape-rendering="geometricpresision"
      style="enable-background: new 0 0 448 512;"
      version="1.1"
      viewBox="0 0 448 512"
      x="0px"
      y="0px"
    >
      <path
        d="m 144.16452,21.032222 h 159.67454 q 123.1748,0 123.1748,128.667868 v 212.64759 q 0,128.66788 -123.1748,128.66788 H 144.16452 q -123.174811,0 -123.174811,-128.66788 V 149.70009 q 0,-128.667868 123.174811,-128.667868 z"
        style="fill: none; fill-opacity: 1; stroke: #969696; stroke-width: 41.96378708; stroke-linejoin: round; stroke-miterlimit: 4; stroke-dasharray: none; stroke-dashoffset: 0; stroke-opacity: 1;"
      />
      <path
        d="M 99.168993,419.0375 H 351.33927 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168993 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z"
      />
    </svg>
  </i>
  <span>
     
    info
     
  </span>
</div>
`;

exports[`Battery component should render correctly LowBattery - low 1`] = `
<div>
  <i
    aria-label="low low"
    class="battery-0-2-1 batteryLow-0-2-3"
    widget-id="low"
    widget-type="Battery"
  >
    <svg
      shape-rendering="geometricpresision"
      style="enable-background: new 0 0 448 512;"
      version="1.1"
      viewBox="0 0 448 512"
      x="0px"
      y="0px"
    >
      <path
        d="m 144.16452,21.032222 h 159.67454 q 123.1748,0 123.1748,128.667868 v 212.64759 q 0,128.66788 -123.1748,128.66788 H 144.16452 q -123.174811,0 -123.174811,-128.66788 V 149.70009 q 0,-128.667868 123.174811,-128.667868 z"
        style="fill: none; fill-opacity: 1; stroke: #969696; stroke-width: 41.96378708; stroke-linejoin: round; stroke-miterlimit: 4; stroke-dasharray: none; stroke-dashoffset: 0; stroke-opacity: 1;"
      />
      <path
        d="M 99.168993,419.0375 H 351.33927 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168993 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z"
      />
    </svg>
  </i>
  <span>
     
    low
     
  </span>
</div>
`;

exports[`Battery component should render correctly MediumBattery - 2 1`] = `
<div>
  <i
    aria-label="2 2"
    class="battery-0-2-1 batteryMedium-0-2-4"
    widget-id="2"
    widget-type="Battery"
  >
    <svg
      shape-rendering="geometricpresision"
      style="enable-background: new 0 0 448 512;"
      version="1.1"
      viewBox="0 0 448 512"
      x="0px"
      y="0px"
    >
      <path
        d="m 144.16452,21.032222 h 159.67454 q 123.1748,0 123.1748,128.667868 v 212.64759 q 0,128.66788 -123.1748,128.66788 H 144.16452 q -123.174811,0 -123.174811,-128.66788 V 149.70009 q 0,-128.667868 123.174811,-128.667868 z"
        style="fill: none; fill-opacity: 1; stroke: #969696; stroke-width: 41.96378708; stroke-linejoin: round; stroke-miterlimit: 4; stroke-dasharray: none; stroke-dashoffset: 0; stroke-opacity: 1;"
      />
      <path
        d="M 99.168857,327.14542 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.19999 v -28 c 0,-6.16001 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.19999 9.698857,11.19999 z M 99.168993,419.0375 H 351.33927 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168993 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z"
        role="img"
      />
    </svg>
  </i>
  <span>
     
    2
     
  </span>
</div>
`;

exports[`Battery component should render correctly MediumBattery - medium 1`] = `
<div>
  <i
    aria-label="medium medium"
    class="battery-0-2-1 batteryMedium-0-2-4"
    widget-id="medium"
    widget-type="Battery"
  >
    <svg
      shape-rendering="geometricpresision"
      style="enable-background: new 0 0 448 512;"
      version="1.1"
      viewBox="0 0 448 512"
      x="0px"
      y="0px"
    >
      <path
        d="m 144.16452,21.032222 h 159.67454 q 123.1748,0 123.1748,128.667868 v 212.64759 q 0,128.66788 -123.1748,128.66788 H 144.16452 q -123.174811,0 -123.174811,-128.66788 V 149.70009 q 0,-128.667868 123.174811,-128.667868 z"
        style="fill: none; fill-opacity: 1; stroke: #969696; stroke-width: 41.96378708; stroke-linejoin: round; stroke-miterlimit: 4; stroke-dasharray: none; stroke-dashoffset: 0; stroke-opacity: 1;"
      />
      <path
        d="M 99.168857,327.14542 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.19999 v -28 c 0,-6.16001 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.19999 9.698857,11.19999 z M 99.168993,419.0375 H 351.33927 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168993 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z"
        role="img"
      />
    </svg>
  </i>
  <span>
     
    medium
     
  </span>
</div>
`;

exports[`Battery component should render correctly MediumBattery - warn 1`] = `
<div>
  <i
    aria-label="warn warn"
    class="battery-0-2-1 batteryMedium-0-2-4"
    widget-id="warn"
    widget-type="Battery"
  >
    <svg
      shape-rendering="geometricpresision"
      style="enable-background: new 0 0 448 512;"
      version="1.1"
      viewBox="0 0 448 512"
      x="0px"
      y="0px"
    >
      <path
        d="m 144.16452,21.032222 h 159.67454 q 123.1748,0 123.1748,128.667868 v 212.64759 q 0,128.66788 -123.1748,128.66788 H 144.16452 q -123.174811,0 -123.174811,-128.66788 V 149.70009 q 0,-128.667868 123.174811,-128.667868 z"
        style="fill: none; fill-opacity: 1; stroke: #969696; stroke-width: 41.96378708; stroke-linejoin: round; stroke-miterlimit: 4; stroke-dasharray: none; stroke-dashoffset: 0; stroke-opacity: 1;"
      />
      <path
        d="M 99.168857,327.14542 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.19999 v -28 c 0,-6.16001 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.19999 9.698857,11.19999 z M 99.168993,419.0375 H 351.33927 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168993 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z"
        role="img"
      />
    </svg>
  </i>
  <span>
     
    warn
     
  </span>
</div>
`;

exports[`Battery component should render correctly NullBatery, default 1`] = `
<div>
  <i
    aria-label=" "
    class="battery-0-2-1 batteryDefault-0-2-2"
    widget-id=""
    widget-type="Battery"
  >
    <svg
      shape-rendering="geometricpresision"
      style="enable-background: new 0 0 448 512;"
      version="1.1"
      viewBox="0 0 448 512"
      x="0px"
      y="0px"
    >
      <path
        d="m 144.16452,21.032222 h 159.67454 q 123.1748,0 123.1748,128.667868 v 212.64759 q 0,128.66788 -123.1748,128.66788 H 144.16452 q -123.174811,0 -123.174811,-128.66788 V 149.70009 q 0,-128.667868 123.174811,-128.667868 z"
        style="fill: none; fill-opacity: 1; stroke: #969696; stroke-width: 41.96378708; stroke-linejoin: round; stroke-miterlimit: 4; stroke-dasharray: none; stroke-dashoffset: 0; stroke-opacity: 1;"
      />
      <path
        d="M 99.168858,143.38516 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698858,11.2 z M 99.168857,235.25069 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z M 99.168857,327.14542 H 351.33914 c 5.33437,0 9.69886,-5.04 9.69886,-11.19999 v -28 c 0,-6.16001 -4.36449,-11.2 -9.69886,-11.2 H 99.168857 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.19999 9.698857,11.19999 z M 99.168993,419.0375 H 351.33927 c 5.33437,0 9.69886,-5.04 9.69886,-11.2 v -28 c 0,-6.16 -4.36449,-11.2 -9.69886,-11.2 H 99.168993 c -5.334371,0 -9.698857,5.04 -9.698857,11.2 v 28 c 0,6.16 4.364486,11.2 9.698857,11.2 z"
      />
    </svg>
  </i>
  <span>
     
     
  </span>
</div>
`;
